dataLayer Builder + Enhanced Ecommerce.tpl

```diff
@@ -0,0 +1,554 @@
+            "displayName": "value",
+            "name": "value",
+            "type": "TEXT",
+            "valueHint": "Variable value"
+          }
+        ],
+        "type": "SIMPLE_TABLE"
+      }
+    ]
+  }
+]
+
+
+___WEB_PERMISSIONS___
+
+[
+  {
+    "instance": {
+      "key": {
+        "publicId": "logging",
+        "versionId": "1"
+      },
+      "param": [
+        {
+          "key": "environments",
+          "value": {
+            "type": 1,
+            "string": "debug"
+          }
+        }
+      ]
+    },
+    "isRequired": true
+  },
+  {
+    "instance": {
+      "key": {
+        "publicId": "access_globals",
+        "versionId": "1"
+      },
+      "param": [
+        {
+          "key": "keys",
+          "value": {
+            "type": 2,
+            "listItem": [
+              {
+                "type": 3,
+                "mapKey": [
+                  {
+                    "type": 1,
+                    "string": "key"
+                  },
+                  {
+                    "type": 1,
+                    "string": "read"
+                  },
+                  {
+                    "type": 1,
+                    "string": "write"
+                  },
+                  {
+                    "type": 1,
+                    "string": "execute"
+                  }
+                ],
+                "mapValue": [
+                  {
+                    "type": 1,
+                    "string": "dataLayer"
+                  },
+                  {
+                    "type": 8,
+                    "boolean": true
+                  },
+                  {
+                    "type": 8,
+                    "boolean": true
+                  },
+                  {
+                    "type": 8,
+                    "boolean": false
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      ]
+    },
+    "clientAnnotations": {
+      "isEditedByUser": true
+    },
+    "isRequired": true
+  }
+]
+
+
+___SANDBOXED_JS_FOR_WEB_TEMPLATE___
+
+const makeTableMap = require('makeTableMap');
+const query = require('queryPermission');
+const createQueue = require('createQueue');
+const dataLayerPush = createQueue('dataLayer');
+const log = require('logToConsole');
+const merge = function() {
+  const obj = {},
+        il = arguments.length;
+  let i = 0,
+      key;
+  for (; i < il; i++) {
+    for (key in arguments[i]) {
+      if (arguments[i].hasOwnProperty(key)) {
+        obj[key] = arguments[i][key];
+      }
+    }
+  }
+  return obj;
+};
+const event = {'event': data.event};
+const currencyParameter = data.currencyParameter;
+let actionFieldParameters = data.actionFieldParameters ? makeTableMap(data.actionFieldParameters, 'name', 'value') : {};
+const actionFieldDropDown = data.actionFieldDropDown;
+ if (actionFieldDropDown === true) {
+    actionFieldParameters = actionFieldParameters;
+ } else {
+   actionFieldParameters = undefined;
+ }
```

Comment:
This is all a bit elaborate when you could replace it with

```
const actionFieldParameterrs = data.actionFieldParameters ? makeTableMap(data.actionFieldParameters, 'name', 'value') : undefined;
```

Reply:
Oh weird, when I went into my custom template I had already switched this statement to shorthand. I must have given you an older version.